tikv/tikv
# add num_deletes in MvccProperties

###  What have you changed?

Add `num_deletes` into `MvccProperties`. It will make GC more accurate.
For compatibility, if a `MvccProperties`  is decoded from a engine which hasn't record `num_deletes`, it will be `num_versions - num_puts`.

###  What is the type of the changes?
Improvement.

###  How is the PR tested?
Manual test
- Create a TiDB cluster with TiKV master branch, insert some data into it.
- Delete some data, and then rolling update TiKV to the branch version.
- TiKV shouldn't panic even if it doesn't record `num_deletes`.
- Delete some data again, and use `tikv-ctl region-properties` to check it works as expected.

Here is a test example result:
For master branch TiKV:
![图片](https://user-images.githubusercontent.com/8407317/77146636-778cb380-6ac6-11ea-8215-3b5488822646.png)
`num_versions` = `num_puts` + `num_deletes`.

For patched TiKV:
![图片](https://user-images.githubusercontent.com/8407317/77146568-4e6c2300-6ac6-11ea-9c49-f2451adffa03.png)
`num_versions` > `num_puts` + `num_deletes`. It's because `num_rollbacks` is not included. So `num_deletes` is more accurate. It will be helpful for https://github.com/tikv/tikv/pull/6728/ .

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

###  Does this PR affect `tidb-ansible`?

###  Refer to a related PR or issue link (optional)

###  Benchmark result if necessary (optional)

###  Any examples? (optional)
